rs/ethereum/cketh/mainnet/minter_upgrade_2024_10_22.md

```diff
@@ -0,0 +1,60 @@
+# Proposal to upgrade the ckETH minter canister
+
+Repository: `https://github.com/dfinity/ic.git`
+
+Git hash: `157c4f6fdfa55b40ebde0fe8aad9aa39ca0dae4a`
+
+New compressed Wasm hash: `fd77fa60cd0cae8d942afd25f77a1778d05357afd297f97859a460a135728ccc`
+
+Target canister: `sv3dd-oaaaa-aaaar-qacoa-cai`
+
+Previous ckETH minter proposal: https://dashboard.internetcomputer.org/proposal/133058
+
+---
+
+## Motivation
+
+This proposal upgrades the ckETH minter to change how Ethereum is reached:
+* To use the [EVM RPC canister](https://dashboard.internetcomputer.org/canister/7hfb6-caaaa-aaaar-qadga-cai) to reach Ethereum. This allows to use a wider variety of Ethereum JSON-RPC providers since most of them require using API keys, which is supported by the [EVM RPC canister](https://dashboard.internetcomputer.org/canister/7hfb6-caaaa-aaaar-qadga-cai).
+* To use a 3-out-of-4 consensus strategy to aggregate answers from multiple providers. This is a more robust strategy than the current one, which requires equality between 3 providers (3-out-of-3).
+
+
+## Upgrade args
+
+```
+git fetch
+git checkout 157c4f6fdfa55b40ebde0fe8aad9aa39ca0dae4a
+cd rs/ethereum/cketh/minter
+didc encode -d cketh_minter.did -t '(MinterArg)' '(variant {UpgradeArg = record {evm_rpc_id = opt principal "7hfb6-caaaa-aaaar-qadga-cai"}})' | xxd -r -p | sha256sum
```

Comment:
The whole line (meaning including the hash at the end) should be
`cbaaaf58254081232b9ce65d738ab8ea822fa3bf1c1f65532059bd40bea775cd`